kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt

```diff
@@ -0,0 +1,250 @@
+              // when selected feature is a point, show editable attributes
+              GeometryType.POINT -> showEditableAttributes(it)
+              // when selected feature is a polyline,
+              GeometryType.POLYLINE -> selectedFeatureIsPolyline = true
+              else -> Log.e(TAG, "Feature of unexpected geometry type selected")
+            }
+            // return, since a feature has been selected
+            return@addDoneListener
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and
+   * 'ST_STR_NAM' attributes.
+   *
+   * @param selectedFeature whose attributes will be edited
+   */
+  private fun showEditableAttributes(selectedFeature: Feature) {
+    // inflate the edit attribute layout
+    val editAttributeView = layoutInflater.inflate(R.layout.edit_attribute_layout, null)
+    // create an alert dialog
+    AlertDialog.Builder(this).apply {
+      setTitle("Edit feature attribute:")
+      setView(editAttributeView)
+      // populate edit texts with current attribute values
+      editAttributeView.addressNumberEditText.setText(selectedFeature.attributes["AD_ADDRESS"].toString())
+      editAttributeView.streetEditText.setText(selectedFeature.attributes["ST_STR_NAM"].toString())
+      setPositiveButton("OK") { _, _ ->
+        // set AD_ADDRESS value to the int from the edit text
+        selectedFeature.attributes["AD_ADDRESS"] =
+          editAttributeView.addressNumberEditText.text.toString().toInt()
+        // set ST_STR_NAM value to the string from edit text
+        selectedFeature.attributes["ST_STR_NAM"] = editAttributeView.streetEditText.text.toString()
+        selectedFeature.featureTable?.updateFeatureAsync(selectedFeature)
+      }
+      setNegativeButton("Cancel") { _, _ -> }
+    }.show()
+  }
+
+  /**
+   * Move the currently selected point feature to the given map point, by updating the selected
+   * feature's geometry and feature table.
+   *
+   * @param mapPoint to which the point geometry should be moved to
+   */
+  private fun movePoint(mapPoint: Point) {
+    // set the selected features' geometry to a new map point
+    selectedFeature?.geometry = mapPoint
+    // update the selected feature's feature table
+    selectedFeature?.featureTable?.updateFeatureAsync(selectedFeature)
+    // clear selection of point
+    clearSelection()
+    selectedFeature = null
+  }
+
+  /**
+   * Move the last of the vertex point of the currently selected polyline to the given map point, by updating the selected
+   * feature's geometry and feature table.
+   *
+   * @param mapPoint to which the last point of the polyline should be moved to
+   */
+  private fun movePolylineVertex(mapPoint: Point) {
+    val polyline = selectedFeature?.geometry as Polyline
+    val polylineBuilder = PolylineBuilder(polyline)
+    // get the first part of the polyline
+    polylineBuilder.parts[0].apply {
```

Comment:
Addressed this in the design but I find this particular usage quite confusing. Is there any way to perhaps move the entire line to some position? That might be too complex